Source/WebCore/Modules/modern-media-controls/controls/adwaita-layout-traits.js

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2021 Apple Inc. All Rights Reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+class AdwaitaLayoutTraits extends LayoutTraits
+{
+    mediaControlsClass()
+    {
+        if (this.isFullscreen)
+            return AdwaitaFullscreenMediaControls;
+        return AdwaitaInlineMediaControls;
+    }
+
+    overridenSupportingObjectClasses()
+    {
+        return null;
+    }
+
+    resourceDirectory()
+    {
+        return "adwaita";
+    }
+
+    controlsAlwaysAvailable()
+    {
+        return this.isFullscreen;
+    }
+
+    controlsNeverAvailable()
+    {
+        return false;
+    }
+
+    supportsIconWithFullscreenVariant()
+    {
+        return this.isFullscreen;
+    }
+
+    supportsDurationTimeLabel()
+    {
+        return true;
+    }
+
+    controlsDependOnPageScaleFactor()
+    {
+        return false;
+    }
+
+    skipDuration()
+    {
+        return 15;
+    }
```

Comment:
This has 15 as duration, then the corresponding icon has 30 written on it. Shouldn't they match?

Reply:
Fixed, I pushed a version of the icons which has “15” written in them 😉